h/views/status.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import logging
+
+from h.util.view import json_view
+from h.exceptions import APIError
+
+
+log = logging.getLogger(__name__)
+
+
+@json_view(route_name='status')
+def status(request):
+    _check_database(request)
+    _check_search(request)
+    _check_realtime(request)
+    return {'status': 'okay'}
+
+
+def _check_database(request):
+    try:
+        request.db.execute('SELECT 1')
+    except Exception as exc:
+        log.exception(exc)
+        raise APIError('Database connection failed')
+
+
+def _check_search(request):
+    try:
+        info = request.es.conn.cluster.health()
+        if info['status'] not in ('green', 'yellow'):
+            raise APIError('Search cluster state is %s' % info['status'])
+    except Exception as exc:
+        log.exception(exc)
+        raise APIError('Search connection failed')
+
+
+def _check_realtime(request):
+    try:
+        request.realtime.connection.release()
+        request.realtime.connection.connect()
```

Comment:
Hmm. I've checked this out locally and I can't reproduce that behaviour: just a call to `connect()` seems to be enough to fail the healthcheck when Rabbit goes away.

That said, just checking we can connect -- while better than nothing -- doesn't tell us all that much. We could, alternatively, check that the return value from `h.celery.celery.control.ping()` is non-empty? That checks that a) we can make a connection, b) Rabbit is processing messages, and c) we have >0 workers running and handling tasks. What do you think?